packages/deployments/contracts/contracts/core/connext/helpers/RelayerProxy.sol

```diff
@@ -366,4 +442,22 @@ contract RelayerProxy is ProposedOwnable, ReentrancyGuard, GelatoRelayFeeCollect
     emit Keep3rChanged(_keep3r, address(keep3r));
     keep3r = IKeep3rV2(_keep3r);
   }
+
+  function _validateProposeSignature(
+    bytes32 _aggregateRoot,
+    uint256 _rootTimestamp,
+    bytes memory _signature
+  ) internal view {
+    // Get the payload
+    bytes32 payload = keccak256(abi.encodePacked(_aggregateRoot, _rootTimestamp));
+    // Recover signer
+    address signer = payload.toEthSignedMessageHash().recover(_signature);
```

Comment:
If this is intended to be deployed in all domains, and some of the allowlisted relayer addresses are the same, then I believe this is subject to signature replay:
1- Call this `proposeAggregateRoot` on Chain A
2- Malicious actor sees the arguments the relayer used (aggregateRoot, rootTimestamp, provided signature)
3- Malicious actor calls `proposeAggregateRoot` on Chain B with the same parameters, which will recover to the same signer as on Chain A.

If I didn't miss anything, a potential fix would be to add "chainId" as a parameter to the function and check to ensure we are on the right chain. We may need history as well or some clever timestamp check to avoid someone submitting the same root, timestamp, and signature twice on the same chain. Adding history, or even adding a nonce would be expensive on mainnet and I feel there are cheaper solutions.